Sources/StreamChatUI/ImageGallery/VideoPlaybackControlView.swift

```diff
@@ -0,0 +1,279 @@
+        .withoutAutoresizingMaskConstraints
+        .withAdjustingFontForContentSizeCategory
+        .withBidirectionalLanguagesSupport
+    
+    /// A label displaying the overall video duration.
+    open private(set) lazy var durationLabel = UILabel()
+        .withoutAutoresizingMaskConstraints
+        .withAdjustingFontForContentSizeCategory
+        .withBidirectionalLanguagesSupport
+    
+    /// A slider used to show a timeline.
+    open private(set) lazy var timeSlider = UISlider()
+        .withoutAutoresizingMaskConstraints
+    
+    /// A container for playback button and time labels.
+    open private(set) lazy var rootContainer = ContainerStackView(axis: .vertical)
+        .withoutAutoresizingMaskConstraints
+    
+    override open func setUp() {
+        super.setUp()
+        
+        timeSlider.minimumValue = 0
+        timeSlider.maximumValue = 1
+        timeSlider.addTarget(self, action: #selector(timeSliderDidChange), for: .valueChanged)
+        
+        timestampLabel.font = appearance.fonts.footnoteBold
+        durationLabel.font = appearance.fonts.footnoteBold
+
+        playPauseButton.addTarget(self, action: #selector(handleTapOnPlayPauseButton), for: .touchUpInside)
+    }
+    
+    override open func setUpLayout() {
+        super.setUpLayout()
+                
+        let bottomContainer = UIView().withoutAutoresizingMaskConstraints
+        
+        bottomContainer.addSubview(timestampLabel)
+        timestampLabel.pin(anchors: [.leading, .top], to: bottomContainer)
+        
+        bottomContainer.addSubview(playPauseButton)
+        playPauseButton.pin(anchors: [.centerX, .top, .bottom], to: bottomContainer)
+        
+        bottomContainer.addSubview(durationLabel)
+        durationLabel.pin(anchors: [.trailing, .top], to: bottomContainer)
+        
+        addSubview(rootContainer)
+        rootContainer.pin(to: self)
+        rootContainer.addArrangedSubview(timeSlider, respectsLayoutMargins: true)
+        rootContainer.addArrangedSubview(bottomContainer, respectsLayoutMargins: true)
+        
+        addSubview(loadingIndicator)
+        loadingIndicator.pin(anchors: [.centerX, .centerY], to: playPauseButton)
+    }
+    
+    override open func setUpAppearance() {
+        super.setUpAppearance()
+        
+        playPauseButton.setTitleColor(.black, for: .normal)
+        timestampLabel.text = DateComponentsFormatter.videoDuration.string(from: 0)
+        durationLabel.text = DateComponentsFormatter.videoDuration.string(from: 0)
+    }
+    
+    override open func updateContent() {
+        super.updateContent()
+        
+        timeSlider.value = .init(content.playingProgress)
+        timestampLabel.text = DateComponentsFormatter.videoDuration.string(from: content.currentTime)
+        durationLabel.text = DateComponentsFormatter.videoDuration.string(from: content.videoDuration)
+        
+        debugPrint("updateContent is main: \(Thread.isMainThread)")
+        
```

Comment:
Don't forget to remove this. Btw why do you need this debugging right now? 🤔

Reply:
We're updating content from player observers so I was checking they are invoked on main thread 🙂